supervisor/api/os.py

```diff
@@ -65,6 +68,15 @@
         vol.Optional(ATTR_SYSTEM_HEALTH_LED): vol.Boolean(),
     }
 )
+
+RE_SWAP_SIZE = re.compile(r"^\d+([KMG](i?B)?|B)?$", re.IGNORECASE)
```

Comment:
So I definitely like this validation better. But I guess one thing for you to consider here is do we need all these options? Just because the OS supports it doesn't necessarily mean we have to provide the full set to our clients. If we think they're unlikely to need anything besides MB and GB we can restrict it to those options for simplicity for example.

I'll leave it up to you. We can always add more options later as people need them but once added we can't take options back without a breaking change.

Reply:
> Just because the OS supports it doesn't necessarily mean we have to provide the full set to our clients.

OTOH if you can set it manually in the `/etc/defaults/` file and its accepted, not accepting back the value that will be shown in GET isn't good either. I prefer to keep support for kilos, as it's what `/proc/meminfo` shows (as `kB`, even though it's technically `KiB`), that's mainly why I implemented this laxly.